modules/gdnative/include/gdnative/gdnative.h

```diff
@@ -286,6 +286,10 @@ void GDAPI godot_print(const godot_string *p_message);
 
 bool GDAPI godot_is_instance_valid(const godot_object *p_object);
 
+//tags used for safe dynamic casting
+void GDAPI *godot_get_class_ptr(const char *p_classname);
+godot_bool GDAPI godot_object_is_class_ptr(const godot_object *p_object, void *p_class_ptr);
```

Comment:
What about `godot_object_is_instance` or `godot_object_is_instance_of` instead ?

The `_ptr` suffix in `godot_object_is_class_ptr` seems to imply we are checking 1rst parameter is somehow a alternative representation of the class pointer provided as 2nd parameter.
Renaming to `godot_object_is_class` doesn't seem right to me either given we an object is an instance of a class and not a directly a class.

Reply:
Is it fine to use a different name than the engine itself uses (just `is_class_ptr`) for GDNative? If so, I do agree `godot_object_is_instance_of` is better.

(`is_class` exists as well - it does the same thing as `is_class_ptr`, but with String comparisons instead of pointer comparisons.)

Reply:
> Is it fine to use a different name than the engine itself uses (just is_class_ptr) for GDNative ?

Ha ! I wasn't [aware of this](https://github.com/godotengine/godot/blob/24e1039eb6fe32115e8d1a62a84965e9be19a2ed/core/object.h#L300-L301) !

I guess my knowledge on the project naming convention falls short here, so I guess sticking to `godot_object_is_class_ptr` is fine.

> is_class exists as well

I think it would be useful to also expose `godot_object_is_class` then ;-)

Reply:
`is_class` is exposed to scripts already, so we could even use it to fix casts without this PR. I'm a bit reluctant to have each cast doing multiple String comparisons - it doesn't use StringName internally for some reason - but that may just be premature optimization.

If we do still want to add a non-String-based safe cast, maybe this naming is clearer?
```cpp
godot_object *godot_object_cast_to(const godot_object *p_object, void *p_class_tag);
```
It does the same thing, just returning `p_object` or `NULL` instead of `true` or `false`. This way, language binding authors don't need to know the details about `is_class_ptr` - the function will only ever be used to reimplement `Object::cast_to`.

Reply:
Having access to `Object::cast_to` directly instead of `is_class_ptr` seems pretty good to me. If we're checking that the object is a class instance via a dynamic cast, we might as well get the casted object instead of just a boolean.

Reply:
The branch has been updated with the improved names and `godot_string_name`. The [C++ binding change](https://github.com/sheepandshepherd/godot-cpp/commit/a809714898ef94fba96db84cc3e2412853432a55) is ready as well - I'll do some more testing before making that PR tomorrow.